lib/core_extensions.rb

```diff
@@ -39,9 +39,17 @@ def initialize(*attribute)
 
     def before_destroy(record)
       klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
-          record.errors.add :base, _("%{record} is used by %{what}") % { :record => record, :what => what }
+        record.association(klass.to_sym).association_scope.each do |what|
+          error_message = _("%{record} is used by %{what}")
+          unless what.is_a? String
+            if record.send(klass).include?(what)
```

Comment:
My problem is finding out if I can use .authorized.
If I do
```
record.class.reflect_on_association(klass.to_sym).klass.respond_to?(:authorized)
```
for klass `:hosts` I get false (because the reflect_on_association returns Host)
and `record.class.reflect_on_association(klass.to_sym).klass. included_modules.include?(Authorizable)`  will be empty
but if I run: `Host::Managed.respond_to?(:authorized)` or `Host::Managed. included_modules.include?(Authorizable)` it will return true.